### PR TITLE
Add errors/warnings to UWP Rendering layer

### PR DESCRIPTION
**Pull request by riarenas**
Parse errors/warning changes will come in separate PR for ease of reviewing. (And because I'm having some issues with it)

### CODE REVIEW

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 1]
```diff
@@ -651,6 +659,16 @@ namespace AdaptiveCards { namespace Uwp
...
             }
+            else
+            {
+                HString elementTypeWrapper;
+                elementTypeWrapper.Set(elementType);
+                std::wstring errorString = L"No Renderer found for type: ";
+                errorString += elementTypeWrapper.GetRawBuffer(nullptr);
+                renderContext->AddError(
+                    FailureStatusCode::NoRendererForType,
+                    HStringReference(errorString.c_str()).Get());
```
Comment (riarenas) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: Taking suggestions for a prettier way to concatenate these strings :(

File: source/uwp/Renderer/lib/AdaptiveFailure.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "AdaptiveFailure.h"
+
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace ABI::AdaptiveCards::Uwp;
+
+namespace AdaptiveCards { namespace Uwp
+{
+
+    HRESULT AdaptiveFailure::RuntimeClassInitialize()
```
Comment (riarenas) [inline source/uwp/Renderer/lib/AdaptiveFailure.cpp]: >AdaptiveFailure [](start = 12, length = 15)

Still looking for a better name than this :(

File: source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl [hunk 3]
```diff
@@ -1945,6 +1976,28 @@ namespace AdaptiveCards
             [default] interface IAdaptiveRenderArgs;
         };
 
+        [
+            uuid(c0222295-bbec-4ff6-bdf3-f207ce78df69),
+            version(NTDDI_WIN10_RS1)
+        ]
+        interface IAdaptiveFailure : IInspectable
```
Comment (riarenas) [inline source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl]: >IAdaptiveFailure [](start = 18, length = 16)

Still looking for a better name.

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 4]
```diff
@@ -651,6 +659,16 @@ namespace AdaptiveCards { namespace Uwp
...
             }
+            else
+            {
+                HString elementTypeWrapper;
+                elementTypeWrapper.Set(elementType);
+                std::wstring errorString = L"No Renderer found for type: ";
+                errorString += elementTypeWrapper.GetRawBuffer(nullptr);
+                renderContext->AddFailure(
+                    FailureStatusCode::NoRendererForType,
+                    HStringReference(errorString.c_str()).Get());
```
Comment (riarenas) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: Any suggestions on how to make this better?

File: source/uwp/Visualizer/ViewModel/DocumentViewModel.cs [hunk 5]
```diff
@@ -179,6 +195,7 @@ public static void InitializeRenderer(AdaptiveHostConfig hostConfig)
 
                 // Custom resource resolvers
                 _renderer.ResourceResolvers.Set("symbol", new MySymbolResourceResolver());
+                _renderer.ElementRenderers.Remove("FactSet");
```
Comment (riarenas) [inline source/uwp/Visualizer/ViewModel/DocumentViewModel.cs]: >_renderer.ElementRenderers.Remove("FactSet"); [](start = 16, length = 45)

Revert. Had this to test the "No renderer for element" warning

File: source/uwp/Renderer/lib/Util.cpp [hunk 6]
```diff
@@ -699,4 +699,10 @@ HRESULT ProjectedElementTypeToHString(ABI::AdaptiveCards::Uwp::ElementType proje
 {
     CardElementType sharedElementType = static_cast<CardElementType>(projectedElementType);
     return UTF8ToHString(CardElementTypeToString(sharedElementType), result);
-}
\ No newline at end of file
+}
+
+bool IsWarning(int FailureStatusCode)
+{
+    int mask = 0xFF000000;
```
Comment (riarenas) [inline source/uwp/Renderer/lib/Util.cpp]: >0xFF000000 [](start = 15, length = 10)

This should be 0xF0000000

File: source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl [hunk 7]
```diff
@@ -1945,6 +1976,28 @@ namespace AdaptiveCards
...
+        {
+            [propget] HRESULT FailureStatusCode([out, retval] FailureStatusCode* value);
+            [propput] HRESULT FailureStatusCode([in] FailureStatusCode value);
```
Comment (riarenas) [inline source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl]: >propput [](start = 13, length = 7)

I think the putters are not needed. I'll see if I need them for once I finish integrating the shared model warnings/errors.

File: source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl (hunk 3 above)
Comment (RebeccaAnne) [reply source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl]: What if separated this out into warnings vs. errors everywhere?  We could make two separate interfaces/enums/add methods/etc. one for warnings and one for errors.  

---
In reply to: [151170456](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151170456) [](ancestors = 151170456)

File: source/uwp/Renderer/lib/XamlBuilder.cpp (hunk 4 above)
Comment (RebeccaAnne) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: Not really, but elementType should be an HString to begin with, then you don't need the extra wrapper variable.

---
In reply to: [151178751](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151178751) [](ancestors = 151178751)

Comment (riarenas) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: Yep, good call.

---
In reply to: [151264474](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151264474) [](ancestors = 151264474,151178751)

File: source/uwp/Renderer/lib/Util.cpp (hunk 6 above)
Comment (riarenas) [reply source/uwp/Renderer/lib/Util.cpp]: Function is no longer needed

---
In reply to: [151202047](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151202047) [](ancestors = 151202047)

File: source/uwp/Renderer/lib/AdaptiveRenderContext.cpp [hunk 8]
```diff
@@ -87,4 +110,6 @@ namespace AdaptiveCards { namespace Uwp
 
         return S_OK;
     }
+
+
```
Comment (RebeccaAnne) [inline source/uwp/Renderer/lib/AdaptiveRenderContext.cpp]: [nit] extra lines

**Pull request merged**